boards/BMS/Src/App/states/App_PreChargeState.c

```diff
@@ -1,14 +1,43 @@
 #include "states/App_AllStates.h"
+#include "states/App_PreChargeState.h"
+#include "App_AccumulatorVoltages.h"
+#include "Io_VoltageSense.h"
+#include "states/App_DriveState.h"
+#include "Io_PreCharge.h"
+#include "states/App_FaultState.h"
+#include <math.h>
+#include "Io_Airs.h"
+#include "states/App_InitState.h"
+#include "Io_Charger.h"
+#include "states/App_ChargeState.h"
 
 #include "App_SetPeriodicCanSignals.h"
 #include "App_SharedMacros.h"
 
+#define S_TO_MS (1000U)
+
+// expected charge time (to 98%) = 4RC * 1000ms/s
+#define PRECHARGE_RESISTANCE (1000U)
+#define PRECHARGE_CAPACITANCE (0.68e-3f)
+#define PRECHARGE_EXPECTED_CHARGE_TIME_MS \
+    ((uint32_t)(PRECHARGE_RESISTANCE * PRECHARGE_CAPACITANCE * S_TO_MS * 4U))
+
+// tracks when the pre-charge begins
+static uint32_t pre_charge_start_ms;
+
 static void PreChargeStateRunOnEntry(struct StateMachine *const state_machine)
 {
     struct BmsWorld *world = App_SharedStateMachine_GetWorld(state_machine);
     struct BmsCanTxInterface *can_tx_interface = App_BmsWorld_GetCanTx(world);
+    struct Clock *            clock            = App_BmsWorld_GetClock(world);
     App_CanTx_SetPeriodicSignal_STATE(
         can_tx_interface, CANMSGS_BMS_STATE_MACHINE_STATE_PRE_CHARGE_CHOICE);
+
+    // capture start time of precharge
+    pre_charge_start_ms = App_SharedClock_GetCurrentTimeInMilliseconds(clock);
```

Comment:
We can use `App_SharedClock_SetPreviousTimeInMilliseconds` to update the clock and remove the `pre_charge_start_ms` entirely